contracts/std-lib-extended/src/bytes.sw

```diff
@@ -354,7 +354,7 @@ fn write_and_read_evm_address(ref mut bytes: Bytes, offset: u64, value: EvmAddre
 fn test_write_and_read_evm_address() {
     let mut bytes = Bytes::with_length(64);
 
-    let value: EvmAddress = EvmAddress::from(0xcafecafecafecafecafecafecafecafecafecafecafecafecafecafecafecafe);
+    let value: EvmAddress = EvmAddress::from(0xdeadeadeadeadeadeadeadeacafecafecafecafecafecafecafecafecafecafe);
```

Comment:
I also made this change as an extra sanity check that only the rightmost bits are kept